source/common/http/conn_manager_impl.cc

```diff
@@ -770,15 +771,26 @@ const Network::Connection* ConnectionManagerImpl::ActiveStream::connection() {
 // can't route select properly without full headers), checking state required to
 // serve error responses (connection close, head requests, etc), and
 // modifications which may themselves affect route selection.
-//
-// TODO(alyssawilk) all the calls here should be audited for order priority,
-// e.g. many early returns do not currently handle connection: close properly.
 void ConnectionManagerImpl::ActiveStream::decodeHeaders(RequestHeaderMapPtr&& headers,
                                                         bool end_stream) {
   ScopeTrackerScopeState scope(this,
                                connection_manager_.read_callbacks_->connection().dispatcher());
   request_headers_ = std::move(headers);
 
+  // Both saw_connection_close_ and is_head_request_ affect local replies: set
+  // them as early as possible.
+  const Protocol protocol = connection_manager_.codec_->protocol();
+  const bool fixed_connection_close =
+      Runtime::runtimeFeatureEnabled("envoy.reloadable_features.fixed_connection_close");
+  if (fixed_connection_close) {
+    state_.saw_connection_close_ =
```

Comment:
nit: s/saw_connection_close_/should_close_connection_ or something since I think the meaning of this over time is now different?